noaa.go

```diff
@@ -33,90 +33,147 @@ type PointsResponse struct {
-	ParentOrganization          string   `json:"parentOrganization"`
-	ResponsibleCounties         []string `json:"responsibleCounties"`
-	ResponsibleForecastZones    []string `json:"responsibleForecastZones"`
-	ResponsibleFireZones        []string `json:"responsibleFireZones"`
-	ApprovedObservationStations []string `json:"approvedObservationStations"`
+	Type                        string         `json:"@type"`
+	URI                         string         `json:"@id"`
+	ID                          string         `json:"id"`
+	Name                        string         `json:"name"`
+	Address                     *OfficeAddress `json:"address"`
+	Telephone                   string         `json:"telephone"`
+	FaxNumber                   string         `json:"faxNumber"`
+	Email                       string         `json:"email"`
+	SameAs                      string         `json:"sameAs"`
+	NWSRegion                   string         `json:"nwsRegion"`
+	ParentOrganization          string         `json:"parentOrganization"`
+	ResponsibleCounties         []string       `json:"responsibleCounties"`
+	ResponsibleForecastZones    []string       `json:"responsibleForecastZones"`
+	ResponsibleFireZones        []string       `json:"responsibleFireZones"`
+	ApprovedObservationStations []string       `json:"approvedObservationStations"`
 }
 
 // StationsResponse holds the JSON values from /points/<lat,lon>/stations
 type StationsResponse struct {
 	Stations []string `json:"observationStations"`
 }
 
+// ForecastElevaion holds the JSON values for a forecast response's elevation.
+type ForecastElevation struct {
+	Value float64 `json:"value"`
+	Units string  `json:"unitCode"`
+}
+
+type ForecastHourlyElevation struct {
+	Value          float64 `json:"value"`
+	Max            float64 `json:"maxValue"`
+	Min            float64 `json:"minValue"`
+	UnitCode       string  `json:"unitCode"`
+	QualityControl string  `json:"qualityControl"`
+}
+
+// Period holds the JSON values for a period within a forecast response's periods.
+type ForecastResponsePeriod struct {
+	ID              int32   `json:"number"`
+	Name            string  `json:"name"`
+	StartTime       string  `json:"startTime"`
+	EndTime         string  `json:"endTime"`
+	IsDaytime       bool    `json:"isDaytime"`
+	Temperature     float64 `json:"temperature"`
+	TemperatureUnit string  `json:"temperatureUnit"`
+	WindSpeed       string  `json:"windSpeed"`
+	WindDirection   string  `json:"windDirection"`
+	Icon            string  `json:"icon"`
+	Summary         string  `json:"shortForecast"`
+	Details         string  `json:"detailedForecast"`
+}
+
+type ForecastResponsePeriodHourly struct {
+	ID               int32   `json:"number"`
+	Name             string  `json:"name"`
+	StartTime        string  `json:"startTime"`
+	EndTime          string  `json:"endTime"`
+	IsDaytime        bool    `json:"isDaytime"`
+	TemperatureTrend *string `json:"temperatureTrend"`
```

Comment:
Any reason to have this be a pointer vs just a string?

Reply:
I noticed this occasionally returns 'null' instead of a string.

Reply:
Testing with a plain string instead of a pointer, if this returns null, you get an empty string.  As such, it doesn't break, so I guess I'll change this to a plain string to match everything else.

This said, IMO, a nil would be faster and easier to test than len(TemperatureTrend) for knowing if there is a temperature trend in the hourly data.

Still, I'm not picky on this point, as I don't really need this information.

Reply:
Makes sense. I'm OK with an empty string (giving up nil check) here since we don't have any guarantees about what the API will send back for any field, really. Thank you for clarifying. Much appreciated.